PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle

```diff
@@ -11,11 +11,36 @@ sourceSets {
 }
 
 dependencies {
-    compile rootProject
-    compile "com.squareup.retrofit2:converter-gson:$retrofit_version"
+    implementation rootProject
+    implementation "com.squareup.retrofit2:converter-gson:$retrofit_version"
     implementation "org.java-websocket:Java-WebSocket:$web_socket_version"
 
     testImplementation rootProject
+
+    // REVIEW: these should come from the rootProject, but somehow don't.
+    //
+    // This is visible e.g. in com/microsoft/intellij/helpers/validator/WebAppValidator.kt:79,
+    // where AzureModel from one of these dependencies is not resolved.
+    //
+    // Any improvements to only have to declare these once would be appreciated.
+    // ---
+    implementation 'com.microsoft.azuretools:azuretools-core:3.50.0', {
+        exclude group: "com.microsoft.azure", module: "azure-client-authentication"
+        exclude group: "com.microsoft.azure", module: "azure-client-runtime"
+        exclude group: "javax.xml.bind", module: "jaxb-api"
+    }
+    implementation 'com.microsoft.azuretools:azure-explorer-common:3.50.0', {
+        exclude group: "com.microsoft.azure", module: "azure-client-authentication"
+        exclude group: "com.microsoft.azure", module: "azure-client-runtime"
+        exclude group: "javax.xml.bind", module: "jaxb-api"
+    }
+    implementation 'com.microsoft.azuretools:hdinsight-node-common:3.50.0', {
+        exclude group: "com.microsoft.azure", module: "azure-client-authentication"
+        exclude group: "com.microsoft.azure", module: "azure-client-runtime"
+        exclude group: "javax.xml.bind", module: "jaxb-api"
+    }
+    implementation group: 'com.microsoft.azure', name: 'azure-auth-helper', version: '0.6.0'
+    // ---
```

Comment:
This shouldn't be here I think, but somehow seems to be the only way to make references work.

Reply:
I have been able to move all of dependencies inside the `mainProjects` block, so it is somewhat resolved.